python/example/single_cell_allen.py

```diff
@@ -0,0 +1,139 @@
+
+    ions = []
+    for kv in fit['conditions'][0]['erev']:
+        region = kv['section']
+        for k, v in kv.items():
+            if k == 'section':
+                continue
+            ion = k[1:]
+            ions.append((region, ion, float(v)))
+
+    return default, regs, ions, mechs
+
+def make_cell(swc, fit, epas_is_param=True):
+    morphology = arbor.load_swc_neuron(swc)
+    # (2) Label the region tags found in the swc with the names used in the parameter fit file.
+    # In addition, label the midpoint of the somarbor.
+    labels = arbor.label_dict({'soma': '(tag 1)',
+                                'axon': '(tag 2)',
+                                'dend': '(tag 3)',
+                                'apic': '(tag 4)',
+                                'midpoint': '(location 0 0.5)'})
+
+    # (3) A function that parses the Allen parameter fit file into components for an arbor.decor
+    dflt, regions, ions, mechanisms = load_allen_fit(fit)
+
+    # (4) Instantiate an empty decor.
+    decor = arbor.decor()
+    # (5) assign global electro-physical parameters
+    decor.set_property(tempK=dflt.tempK, Vm=dflt.Vm, cm=dflt.cm, rL=dflt.rL)
+    # (6) override regional electro-physical parameters
+    for region, vs in regions:
+        decor.paint(f'"{region}"', tempK=vs.tempK, Vm=vs.Vm, cm=vs.cm, rL=vs.rL)
+    # (7) set reversal potentials
+    for region, ion, e in ions:
+        decor.paint(f'"{region}"', ion_name=ion, rev_pot=e)
+    decor.set_ion('ca', int_con=5e-5, ext_con=2.0, method=arbor.mechanism('nernst/x=ca'))
+    # (8) assign ion dynamics
+    for region, mech, values in mechanisms:
+        nm = mech
+        vs = {}
+        sp = '/'
+        for k, v in values.items():
+            # pas::e became a global between .5 and .6, toggle epas_is_param
+            if epas_is_param and mech == 'pas' and k == 'e':
+                nm = f'{nm}{sp}{k}={v}'
+                sp = ','
+            else:
+                vs[k] = v
+        decor.paint(f'"{region}"', arbor.density(arbor.mechanism(nm, vs)))
+    # (9) attach stimulus and spike detector
+    decor.place('"midpoint"', arbor.iclamp(200, 1000, 0.15), 'ic')
+    decor.place('"midpoint"', arbor.spike_detector(-40), 'sd')
+    # (10) discretisation strategy: max compartment length
+    decor.discretization(arbor.cv_policy_max_extent(20))
+
+    # (11) Create cell
+    return arbor.cable_cell(morphology, labels, decor)
+
+# (12) Create cell, model
+cell = make_cell('single_cell_allen.swc', 'single_cell_allen_fit.json')
+model = arbor.single_cell_model(cell)
+
+# (13) Set the probe
+model.probe('voltage', '"midpoint"', frequency=200)
+
+# (14) Install the Allen mechanism catalogue.
+model.properties.catalogue.extend(arbor.allen_catalogue(), "")
+
+# (15) Run simulation
+model.run(tfinal=1400, dt=0.005)
+
+# (16) Load and scale reference
+reference = 1000.0*pandas.read_csv('single_cell_allen_neuron_ref.csv')['U/mV'].values[:-1] - 14.0
```

Comment:
You can load or at least motivate the -14.

Reply:
Actually, we should load it from the file.